py/vttest/local_database.py

```diff
@@ -50,9 +54,14 @@ def vtgate_addr(self):
 
   def config(self):
     """Returns a dict with enough information to be able to connect."""
-    return {
-        'port': vt_processes.vtgate_process.port,
-        }
+    if self.mysql_only:
+      return {
+          'socket': self.mysql_db.unix_socket(),
```

Comment:
OK cool. Go ahead and merge this if you want to move forward...

(my feeling is you'll need mysql_db to have a config_json() or something method that returns the dict, so the mysqlctl one can return exactly what you have there, and the google3 can return a port. Both should return also a username and password. Then your unit tests can just read these values and pass them to the db conn layer)